src/main/java/com/puppycrawl/tools/checkstyle/gui/ParseTreeTablePresentation.java

```diff
@@ -75,7 +74,7 @@ public ParseTreeTablePresentation(DetailAST parseTree) {
      * @param parseTree DetailAST parse tree.
      */
     protected final void setParseTree(DetailAST parseTree) {
-        ((AST) root).setFirstChild(parseTree);
+        ((DetailAST) root).setFirstChild(parseTree);
```

Comment:
This code has to be reverted because of:
> ParseTreeTablePresentation.java (2)
77: setParseTree() Cast to DetailAST can be weakened to 'AST' 
188: getChildCount() Cast to DetailAST can be weakened to 'AST'